OMGVA PoS/ClientApp/src/pages/SelectBusinessPage.tsx

```diff
@@ -0,0 +1,65 @@
+﻿import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom'
+import { getAllBusinesses } from '../services/businessService';
+import ClickableListItem from '../components/List/ClickableListItem';
+import '../index.css';
+import '../components/List/ClickableListItem.css';
+import { getTokenRole } from '../utils/tokenUtils';
+
+interface SelectBusinessPageProps {
+    token : string | null
+}
+
+const SelectBusinessPage: React.FC<SelectBusinessPageProps> = ({token: authToken}) => {
+    const [listItems, setListItems] = useState<Array<JSX.Element>>();
+    const [error, setError] = useState<string | null>(null);
+    const navigate = useNavigate();
+
+    const getBusinesses = async () => {
+        setError(null);
+
+        try {
+            const { result, error } = await getAllBusinesses(authToken);
+
+            if (!result) {
+                setError('Problem acquiring businesses: ' + error);
+            }
+            else {
+                setListItems(result.map(business =>
+                    <ClickableListItem key={business.Id} text={business.Name} url={'/business/' + business.Id}/>));
+            }
+        } catch (err: any) {
+            setError(err.message || 'An unexpected error occurred.');
+        }
+    }
+
+    useEffect(() => {
+        if (authToken) {
+            const role = getTokenRole(authToken);
+            if (role !== "Admin") {
+                navigate('/');
+                return;
+            }
+            getBusinesses();
+        }
+        else {
+            setError("Authorization token is missing!");
```

Comment:
This seems to technical for a client, no? Shouldn't it be more like "You have to authenticate first"?

Reply:
Fair point

Reply:
Changed it.